ph5/clients/ph5availability.py

```diff
@@ -247,20 +273,61 @@ def get_availability_extent(self, station='*', location='*',
 
                         ph5_das = st['das/serial_number_s']
                         ph5_channum = st['channel_number_i']
-
-                        ret = self.get_time(st, starttime, endtime)
-                        if ret == -1:
-                            continue
-                        ph5_starttime, ph5_endtime = ret
+                        ph5_sample_rate = self.get_sample_rate(st)
 
                         earliest, latest = self.ph5.get_extent(
-                            ph5_das, ph5_channum, ph5_starttime, ph5_endtime)
-                        tup = (ph5_seed_station,
-                               ph5_loc, ph5_channel, earliest, latest)
+                            ph5_das, ph5_channum, ph5_sample_rate,
+                            starttime, endtime)
+                        if earliest is None:
+                            continue
+                        latest = round(latest, 6)
+                        if not include_sample_rate:
+                            tup = (ph5_seed_station, ph5_loc, ph5_channel,
+                                   earliest, latest)
+                        else:
+                            tup = (ph5_seed_station, ph5_loc, ph5_channel,
+                                   earliest, latest, ph5_sample_rate)
                         availability_extents.append(tup)
 
         return availability_extents
 
+    def get_das_avail(self, das_t, samplerate, earliest, latest, start, end):
```

Comment:
a get_availability at the das level already exists in ph5api. Cutting data is too slow. There is no need to actually grab data.  This could lead to memory issues on large requests and make things too slow to work